### PR TITLE
jsonrpc: Support up to 2 return values

### LINKED ISSUE #933: Allow for less than two return values in RPC handlers
Right now RPC handlers must have two return values, and the second return value must be of type `*jsonrpc.Error`. It would convenient if the following three cases were supported:

- zero return values
- one return value (response or error)
- two return values, where the second must be an error <- we only support this right now

This is very similar to geth's RPC server [semantics](https://github.com/ethereum/go-ethereum/blob/7f756dc1185d7f1eeeacb1d12341606b7135f9ea/rpc/doc.go#L29-L30).

### PR DESCRIPTION
**Pull request by Exca-DK**
closes #933

### CODE REVIEW

File: jsonrpc/server_test.go [hunk 1]
```diff
@@ -453,6 +493,22 @@ func TestHandle(t *testing.T) {
...
+			req: `{"jsonrpc": "2.0", "method": "echoValue", "params": [10], "id": 1}`,
+			res: `{"jsonrpc":"2.0","result":10,"id":1}`,
+		},
+		"echo error": {
+			req: `{"jsonrpc": "2.0", "method": "echoError", "params": ["echo"], "id": 1}`,
+			res: `{"jsonrpc":"2.0","error":{"code":44,"message":"echo"},"id":1}`,
+		},
+		"echo nil error": {
+			req: `{"jsonrpc": "2.0", "method": "echoError", "params": ["nil"], "id": 1}`,
+			res: `{"jsonrpc":"2.0","id":1}`,
```
Comment (joshklop) [inline jsonrpc/server_test.go:506]: I believe `result` is a required member of the response. ([spec](https://www.jsonrpc.org/specification#response_object))

Comment (Exca-DK) [reply jsonrpc/server_test.go]: You are right, [result should be null](https://github.com/ethereum/go-ethereum/blob/master/rpc/testdata/reqresp-noargsrets.js)

**Comment by omerfirmak** [comment]
I am not sure I am following, what is the point?

**Comment by joshklop** [comment]
> I am not sure I am following, what is the benefit?

Described here https://github.com/NethermindEth/juno/issues/933

**Comment by omerfirmak** [comment]
> > I am not sure I am following, what is the benefit?
> 
> Described here #933

"if it ain't broke, don't fix it" - someone wise, probably

Unless we have an immediate use for the feature or cleaning up technical dept that is making our lives harder; I don't really see the point in introducing things like this.

**Comment by omerfirmak** [comment]
> > > I am not sure I am following, what is the benefit?
> > 
> > 
> > Described here #933
> 
> "if it ain't broke, don't fix it" - someone wise, probably
> 
> Unless we have an immediate use for the feature or cleaning up technical dept that is making our lives harder; I don't really see the point in introducing things like this.

Because at the end of the day, we will end up maintaining it for little-to-no benefit on our account.

"best code is no code at all" - someone even wiser

**Comment by Exca-DK** [comment]
If this is no longer wanted the issue and pr can be closed :sweat_smile:

**Pull request closed without merging**